cmd/tuf/createsecret/main.go

```diff
@@ -0,0 +1,159 @@
+// Copyright 2022 The Sigstore Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"bytes"
+	"flag"
+	"os"
+
+	"github.com/sigstore/cosign/cmd/cosign/cli/rekor"
+	corev1 "k8s.io/api/core/v1"
+	apierrs "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"knative.dev/pkg/logging"
+	"knative.dev/pkg/signals"
+	"sigs.k8s.io/release-utils/version"
+)
+
+const (
+	// These are the keys held in the secrets
+	fulcioSecretKey = "cert"
+	ctSecretKey     = "public"
+
+	// These are the fields we create in the secret
+	fulcioSecretKeyOut = "fulcio-cert"
+	ctSecretKeyOut     = "ct-pubkey"
+	rekorSecretKeyOut  = "rekor-pubkey"
+)
+
+var (
+	fulcioSecret = flag.String("fulcio-secret", "fulcio-secret", "Secret holding Fulcio cert")
+	// URL to Rekor to query for the Rekor public key to include in the trust root.
+	// Could replace with rekor-pubkey, if that can be resolved early enough (don't know)
+	rekorURL = flag.String("rekor-url", "http://rekor.rekor-system.svc", "Address of the Rekor server")
+	// TODO(vaikas): Consider creating this secret possibly.
+	/*
+		rekorSecret   = flag.String("rekor-secret", "fulcio-secrets", "Secret holding Fulcio cert")
+	*/
+	ctlogSecret = flag.String("ctlog-secret", "ctlog-public-key", "Secret holding CTLog public key")
+
+	secretName = flag.String("secret-name", "tuf-secrets", "Name of the secret to create holding necessary information to create/run tuf service")
+)
+
+func main() {
+	flag.Parse()
+	ns := os.Getenv("NAMESPACE")
+	if ns == "" {
+		panic("env variable NAMESPACE must be set")
+	}
+	ctx := signals.NewContext()
+
+	versionInfo := version.GetVersionInfo()
+	logging.FromContext(ctx).Infof("running create_secrets Version: %s GitCommit: %s BuildDate: %s", versionInfo.GitVersion, versionInfo.GitCommit, versionInfo.BuildDate)
+
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		logging.FromContext(ctx).Panicf("Failed to get InClusterConfig: %v", err)
+	}
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		logging.FromContext(ctx).Panicf("Failed to get clientset: %v", err)
+	}
+
+	// Grab the Fulcio root cert
+	nsSecrets := clientset.CoreV1().Secrets(ns)
```

Comment:
shouldn't the namespace here be the fulcio-system? the env will be the tuf-system.

Reply:
Yeah, I'll be doing some cleanup of those, I tried to break this into separate steps to make it easier to start with :)

Reply:
Oh! Okay the `createsecret` job just sets up the secret data, without populating it? 

I got this error:
```
{"level":"panic","ts":1659373304.8090189,"logger":"fallback","caller":"createsecret/main.go:86","msg":"Fulcio cert key \"cert\" is missing tuf-system/fulcio-secret","stacktrace":"main.main\n\tgithub.com/sigstore/scaffolding/cmd/tuf/createsecret/main.go:86\nruntime.main\n\truntime/proc.go:250"}
panic: Fulcio cert key "cert" is missing tuf-system/fulcio-secret
```

because it seemed like the job was trying to make sure the secret was non-empty. but the secret seems to be copied into the `tuf-system` after `createsecret` is complete

Reply:
Actually, you can't grant access to only specific secrets in namespaces. Since we need to peek at two namespaces, ctlog-system / fulcio-system, I thought it would be better to have a step which creates the 'mountable' secret that the server then uses.
So, this one assumes the copying of the secrets outside of the process (there's the kubectl shell there that copies them into this namespace), and fetches the Rekor root and pulls out only the pieces we care about in the server.